platforms/cuda/src/CudaTorchKernels.cpp

```diff
@@ -173,26 +181,33 @@ void CudaCalcTorchForceKernel::addForces(torch::Tensor& forceTensor) {
 }
 
 /**
- * This function launches  the workload in a way  compatible with CUDA
- * graphs as far  as OpenMM-Torch goes.  Capturing  this function when
- * the model  is not  itself graph compatible  (due to,  for instance,
+ * This function launches the workload in a way compatible with CUDA
+ * graphs as far as OpenMM-Torch goes.  Capturing this function when
+ * the model is not itself graph compatible (due to, for instance,
  * implicit synchronizations) will result in a CUDA error.
  */
 static void executeGraph(bool outputsForces, bool includeForces, torch::jit::script::Module& module, vector<torch::jit::IValue>& inputs, torch::Tensor& posTensor, torch::Tensor& energyTensor,
-                         torch::Tensor& forceTensor) {
+                         torch::Tensor& forceTensor, map<string, torch::Tensor>& derivInputs) {
+    vector<torch::Tensor> gradInputs;
+    if (!outputsForces)
+        gradInputs.push_back(posTensor);
+    for (auto& deriv : derivInputs)
+        gradInputs.push_back(deriv.second);
+    auto none = torch::Tensor();
     if (outputsForces) {
         auto outputs = module.forward(inputs).toTuple();
         energyTensor = outputs->elements()[0].toTensor();
         forceTensor = outputs->elements()[1].toTensor();
+        if (gradInputs.size() > 0)
+            energyTensor.backward(none, false, false, gradInputs);
     } else {
         energyTensor = module.forward(inputs).toTensor();
         // Compute force by backpropagating the PyTorch model
         if (includeForces) {
             // CUDA graph capture sometimes fails if backwards is not explicitly requested w.r.t positions
-	    // See https://github.com/openmm/openmm-torch/pull/120/
-	    auto none = torch::Tensor();
-	    energyTensor.backward(none, false, false, posTensor);
-	    // This is minus the forces, we change the sign later on
+            // See https://github.com/openmm/openmm-torch/pull/120/
+            energyTensor.backward(none, false, false, gradInputs);
+            // This is minus the forces, we change the sign later on
             forceTensor = posTensor.grad().clone();
             // Zero the gradient to avoid accumulating it
             posTensor.grad().zero_();
```

Comment:
You need to zero the grads of the parameter derivatives I believe. Since the tensors are static they would get accumulated otherwise.